eng/scripts/dependencies/compare-deps-files.ps1

```diff
@@ -0,0 +1,53 @@
+param(
+  [string]$PSDepsFile,
+  [string]$AzSdkDepsFile
+)
+
+function Get-DepsFileContent($depsfile)
+{
+  if ($depsfile.StartsWith("http")) {
+    try {
+      $response = Invoke-WebRequest -Uri $depsfile
+      $content = $response.Content
+    }
+    catch {
+      $statusCode = $_.Exception.Response.StatusCode.value__
+      Write-Error "Failed to download deps file [$statusCode] $depsfile"
+    }
+  }
+  elseif (Test-Path $depsFile) {
+    $content = Get-Content $depsFile
+  }
+  else {
+    Write-Error "File $depsFile does not exists"
+  }
+
+  return $content
+}
+
+$psDeps = Get-DepsFileContent $PSDepsFile | ConvertFrom-Json
+$azDeps = Get-DepsFileContent $AzSdkDepsFile | ConvertFrom-Json
+
+$psDepsVersion = @{}
+$psDeps.libraries.PSObject.Properties | ForEach-Object { $n, $v = ($_.Name -split "/"); $psDepsVersion[$n] = $v }
+
+$azDepsVersions = @{}
+$azDeps.libraries.PSObject.Properties | ForEach-Object { $n, $v = ($_.Name -split "/"); $azDepsVersions[$n] = $v }
+
+foreach ($key in $azDepsVersions.Keys)
+{
+  if ($psDepsVersion.Contains($key)) {
+
+    $psVer = $psDepsVersion[$key]
+    $azVer = $azDepsVersions[$key]
+
+    Write-Verbose "[$key] PSDepsFile has version [$psVer] and AzSdkDepsFile has version [$azVer]"
+    
+    # We should use proper version compare instead of string compare
```

Comment:
Just cast to `[Version]`, e.g. `[version]"1.2.3.4"`. The `-gt` operator works fine then. But if these have semver prerelease or metadata, it won't work. You'll want to trim/ignore any thing after and including "-" or "+".

Reply:
I'm doing that now that they are assembly versions. Before they were package sem versions which didn't work.